strawberry/experimental/pydantic/fields.py

```diff
@@ -64,13 +60,22 @@
 }
 
 
-def get_basic_type(type_):
+def get_basic_type(type_) -> Type[Any]:
     if isinstance(type_, type):
         if issubclass(type_, pydantic.ConstrainedInt):
             return int
         if issubclass(type_, pydantic.ConstrainedStr):
             return str
 
+    if hasattr(type_, "__name__"):
+        if "ConstrainedListValue" in type_.__name__:
+            # pydantic ConstrainedListValue is created dynamically
+            # it isn't wrapped in a list, so we need to do it ourselves
+            return List[get_basic_type(type_.item_type)]  # type: ignore
```

Comment:
mypy complains about List not being indexable with a type? which is weird

Reply:
yeah, don't think mypy is able to understand this code 😊